src/main/java/com/wolox/training/controller/UserController.java

```diff
@@ -95,7 +107,9 @@ public User update(
             throw new UserIdMismatchException("Id doesn't not match");
         }
 
-        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException("User not found"));
+        User u = userRepository.findById(id).orElseThrow(() -> new UserNotFoundException("User not found"));
+        user.setPassword(u.getPassword());
```

Comment:
You have to encode it at updating too.

Reply:
Maybe you could move the encryption to the domain setter

Reply:
The update service keep the encoded password and not permit update password in this resource, the encode process for password is alocated in create user service. This is a good work  ?

Reply:
Ok, I get it now :stuck_out_tongue: